src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -427,5 +529,178 @@ private ImmutableArray<BoundExpression> BindInterpolatedStringParts(BoundUnconve
+            }
+
+            // We need to find the appropriate argument expression for every expected parameter, and error on any that occur after the current parameter
+
+            ImmutableArray<int> handlerArgumentIndexes;
+
+            if (memberAnalysisResult.ArgsToParamsOpt.IsDefault && arguments.Count == parameters.Length)
+            {
+                // No parameters are missing and no remapped indexes, we can just use the original indexes
+                handlerArgumentIndexes = handlerParameterIndexes;
+            }
+            else
+            {
+                // Args and parameters were reordered via named parameters, or parameters are missing. Find the correct argument index for each parameter.
+                var handlerArgumentIndexesBuilder = ArrayBuilder<int>.GetInstance(handlerParameterIndexes.Length, fillWithValue: BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter);
+                for (int argumentIndex = 0; argumentIndex < arguments.Count; argumentIndex++)
+                {
+                    // The index in the original parameter list we're looking to match up.
+                    var argumentParameterIndex = memberAnalysisResult.ParameterFromArgument(argumentIndex);
+                    for (int handlerParameterIndex = 0; handlerParameterIndex < handlerParameterIndexes.Length; handlerParameterIndex++)
+                    {
+                        // Is the original parameter index of the current argument the parameter index that was specified in the attribute?
+                        if (argumentParameterIndex == handlerParameterIndexes[handlerParameterIndex])
+                        {
+                            // We can't just bail out on the first match: users can duplicate parameters in attributes, causing the same value to be passed twice.
+                            Debug.Assert(handlerArgumentIndexesBuilder[handlerParameterIndex] == BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter);
+                            handlerArgumentIndexesBuilder[handlerParameterIndex] = argumentIndex;
+                        }
+                    }
+                }
+
+                handlerArgumentIndexes = handlerArgumentIndexesBuilder.ToImmutableAndFree();
+            }
+
+            var argumentPlaceholdersBuilder = ArrayBuilder<BoundInterpolatedStringArgumentPlaceholder>.GetInstance(handlerArgumentIndexes.Length);
+            var argumentRefKindsBuilder = ArrayBuilder<RefKind>.GetInstance(handlerArgumentIndexes.Length);
+            bool hasErrors = false;
+
+            // Now, go through all the specified arguments and see if any were specified _after_ the interpolated string, and construct
+            // a set of placeholders for overload resolution.
+            for (int i = 0; i < handlerArgumentIndexes.Length; i++)
+            {
+                int argumentIndex = handlerArgumentIndexes[i];
+                Debug.Assert(argumentIndex != interpolatedStringArgNum);
```

Comment:
>Debug.Assert(argumentIndex != interpolatedStringArgNum); [](http://example.com/codeflow?start=16&length=56)

Is this validated when we crack the attribute?

Reply:
Yes: [https://github.com/dotnet/roslyn/blob/features/interpolated-string/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs#L781](https://github.com/dotnet/roslyn/blob/features/interpolated-string/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs#L781). We return default and set the error flag for that case.